influxql/iterator.go

```diff
@@ -259,9 +261,11 @@ func newAuxIteratorFields(seriesKeys SeriesList, opt IteratorOptions) auxIterato
 
 func (a auxIteratorFields) close() {
 	for _, f := range a {
+		f.mtx.Lock()
 		for _, itr := range f.itrs {
 			itr.Close()
 		}
+		f.mtx.Unlock()
```

Comment:
Adding a `auxIteratorField.Close()` serves a couple purposes:

1. You can defer the `f.mtx.Unlock()` so that if a child iterator panics it'll still unlock the mutex.

2. It encapsulates the functionality for `auxIteratorField`. It's a small function right now but it may become more complicated in the future (or called from another location).

Also, I just realized, can you rename the mutex from `mtx` to `mu`. That's been our convention in the codebase so far.